src/rpc/core_rpc_server.cpp

```diff
@@ -2965,13 +2965,34 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
-  bool core_rpc_server::on_storage_server_ping(const COMMAND_RPC_STORAGE_SERVER_PING::request&,
+  bool core_rpc_server::on_storage_server_ping(const COMMAND_RPC_STORAGE_SERVER_PING::request& req,
                                                COMMAND_RPC_STORAGE_SERVER_PING::response& res,
                                                epee::json_rpc::error&,
                                                const connection_context*)
   {
-    m_core.m_last_storage_server_ping = time(nullptr);
-    res.status = "OK";
+
+    const uint8_t hf_version = m_core.get_hard_fork_version(m_core.get_current_blockchain_height());
+
+    const int cur_version[] = { req.version_major, req.version_minor, req.version_patch };
+
+    const bool update_needed = std::lexicographical_compare(cur_version,
+                                                            cur_version + 3,
+                                                            service_nodes::MIN_STORAGE_SERVER_VERSION,
+                                                            service_nodes::MIN_STORAGE_SERVER_VERSION + 3);
+
+    if (hf_version >= network_version_13 && update_needed) {
```

Comment:
I think we should print this warning even if it's before v13, then just add a "(this will be enforced starting at the next fork)" if before v13.

Reply:
Updated to enforce immediately (changing min version to 1.0.5).